### ISSUE TITLE
Allow to run multiple tasks concurrently

### ISSUE THREAD

**Issue by whitecolor** [Microsoft/vscode#7863]
Didn't find this issue. Allow to run multiple tasks concurrently. Currently it ask s to terminate running one.

This actually for watching tasks. And also seem that tasks are run some way differently that upcoming workbench terminal as they do not keep colors inside running scripts. Terminal and task runner seem to be very relative technically.

**Comment by dbaeumer** [comment]
Dups #981

**Comment by whitecolor** [comment]
@dbaeumer 
I've seen this issue, I thought that the question was a little bit different, it is about defining multiple tasks, a this issue is about concurrent execution, including tasks that are supposed to run "forever" like server/watching/etc.

**Comment by dbaeumer** [comment]
The reason why I folded them is that if we allow running multiple tasks supporting multiple commands is almost a must. That we we plan to solve them together in one go.

**Comment by twoeo** [comment]
I want to use 2 tasks concurrently:
- tsc watch
- tslint watch

They have different problemMatchers, so can not define as one task.

**Comment by dbaeumer** [comment]
@twoeo this is possible with the task 2.0.0. It can run multiple tasks in parallel.

**Comment by ceituna** [comment]
I think what @k8w meant and what I'm facing is starting multiple tasks in one command. I now have two tasks that I want to start at build, but I can only choose one default task. The other one I have to start independently via the Command Palette or by choosing it through Tasks -> Run Task. If I set both tasks as default build tasks with `"group": {
                "kind": "build",
                "isDefault": true
            },` they don't start automatically after using Ctrl+Shift+B and I have to choose one and repeat the process and choose the other.

**Comment by dbaeumer** [comment]
@ceituna are your tasks watching tasks. If not you can define a new tasks that depends on the two others and mark that one as the default build task. However this is currently not possible if they are watching tasks.

**Comment by ceituna** [comment]
@dbaeumer Only one of them is (a tsc watch). I wasn't aware of the dependsOn option. Couldn't find it in the docs. I've now setup a new task which depends on the other two as the default build task and everything works as expected. Thanks for the hint!

**Comment by dbaeumer** [comment]
You are welcome. Agree that this should be documented as well.
